lib/util/package-rules/repositories.ts

```diff
@@ -0,0 +1,47 @@
+import is from '@sindresorhus/is';
+import { minimatch } from 'minimatch';
+import type { PackageRule, PackageRuleInputConfig } from '../../config/types';
+import { logger } from '../../logger';
+import { regEx } from '../regex';
+import { Matcher } from './base';
+import { massagePattern } from './utils';
+
+export class RepositoriesMatcher extends Matcher {
+  override matches(
+    { repository }: PackageRuleInputConfig,
+    { matchRepositories }: PackageRule
+  ): boolean | null {
+    if (is.undefined(matchRepositories)) {
+      return null;
+    }
+
+    if (is.undefined(repository)) {
+      return false;
+    }
+
+    return matchRepositories.some(
+      (matchRepositoryName) =>
+        isRegexAndMatches(matchRepositoryName, repository) ||
+        minimatch(repository, matchRepositoryName, { dot: true })
+    );
+  }
+}
+
```

Comment:
I would like to see an `exclude` here too. I can imagine many scenarios, I need to apply configs to everyrepo except one or two.

Reply:
Copy that.  Added `excludeRepositories` as an option